source/transfer_list.rst

```diff
@@ -129,35 +124,64 @@ The TE header is defined in :numref:`tab_te_header`.
      - Description
 
    * - tag_id
-     - 0x4
+     - 0x3
      - 0x0
      - The entry type identifier.
 
    * - hdr_size
-     - 0x4
-     - 0x4
-     - The size of this entry header in bytes.
+     - 0x1
+     - 0x3
+     - The size of this entry header in bytes. This field is set to 8 for the TE header layout described in this version of the table.
 
    * - data_size
      - 0x4
-     - 0x8
-     - The size of the data content in bytes.
-
-   * - reserved
      - 0x4
-     - 0xc
-     - Reserved, must be zero.
+     - The exact size of the data content in bytes, not including inter-TE padding.
 
 
-Entry type ranges
+Entry type allocation
 -----------------
 
-The content of the data section is determined by the tag id. The tag id space contains two ranges:
-
- #. Standard range, and
- #. Non-standard range
-
-The *tag_id* ranges are described in :numref:`tab_tag_id_ranges`.
+Tag IDs must be allocated in this specification before use. A new tag ID can be
+allocated by submitting a pull request to this repository that adds a
+description of the respective TE data layout to this specification. Tag IDs do
+not have to be allocated in order. Submitters are encouraged to try to group
+tag IDs together in logical clusters at 256-aligned boundaries (e.g. all tags
+related to a particular chipset or to a particular firmware project could use
+adjacent tag numbers), but there are no predefined ranges and no reservations
+of tag ranges for specific use.
+
+Tags are expected to have a simple layout (representable by a C structure) and
+each tag should only represent data for a single logical concept. Data for
+multiple distinct concepts should be split across different tags, even if
+they're always expected to appear together on the first platform adding the tag
+(to encourage reusability in different situations). The same tag ID may occur
+multiple times in the TL to represent multiple instances of the same kind of
+object. Tag layouts (including the meaning of all fields) are considered stable
+after being added to this specification and may never be changed in a backwards
+incompatible way. If a backwards-incompatible change is desired, a new tag ID
+should be allocated for the new version of the layout instead.
+(Backwards-compatible ways to change existing layouts could be increasing the
+data_length to append new fields behind the structure or using fields in the
+layout that were previously marked as reserved, as long as the TE would still
+be considered valid for older implementations that ignore these new fields. TE
+layouts which have been changed like that must clearly document which fields
+have been added after the first version.)
```

Comment:
Added an extra blurb about tag stability / backwards-compatibility here in the latest push, which I thought was also important to clearly define.